buildenv/jenkins/docker-slaves/ppc64le/centos7/Dockerfile

```diff
@@ -0,0 +1,246 @@
+  && cpanm Text::CSV \
+  && cpanm JSON
+
+# Setup links to GCC-4.8.5
+RUN rm /usr/bin/c++ /usr/bin/cc \
+  && ln -s g++ /usr/bin/c++ \
+  && ln -s gcc /usr/bin/cc
+
+# Install GCC-7.3.1
+RUN yum -y install environment-modules \
+  && source /etc/profile.d/modules.sh \
+  && wget ftp://ftp.unicamp.br/pub/linuxpatch/toolchain/at/redhat/RHEL7/gpg-pubkey-6976a827-5164221b \
+  && rpm --import gpg-pubkey-6976a827-5164221b \
+  && echo "# Begin of configuration file" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "[at11.0]" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "name=Advance Toolchain Unicamp FTP" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "baseurl=ftp://ftp.unicamp.br/pub/linuxpatch/toolchain/at/redhat/RHEL7" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "failovermethod=priority" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "enabled=1" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "gpgcheck=1" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "gpgkey=ftp://ftp.unicamp.br/pub/linuxpatch/toolchain/at/redhat/RHEL7/gpg-pubkey-6976a827-5164221b" >> /etc/yum.repos.d/at11.0.repo \
+  && echo "# End of configuration file" >> /etc/yum.repos.d/at11.0.repo \
+  && yum -y install advance-toolchain-at11.0-runtime advance-toolchain-at11.0-devel advance-toolchain-at11.0-perf
+
+#Building and setting up git version 2.5.3
+RUN yum -y install curl-devel expat-devel gettext-devel openssl-devel zlib-devel perl-ExtUtils-MakeMaker \
+  && cd /usr/src \
+  && wget https://www.kernel.org/pub/software/scm/git/git-2.5.3.tar.gz \
+  && tar xzf git-2.5.3.tar.gz \
+  && rm git-2.5.3.tar.gz \
+  && cd git-2.5.3 \
+  && make prefix=/usr/local/git all \
+  && make prefix=/usr/local/git install \
+  && rm -f /usr/bin/git \
+  && ln -s /usr/local/git/bin/git /usr/bin/git
+
+# Install ant version 1.10.5.
+RUN wget http://www.us.apache.org/dist/ant/binaries/apache-ant-1.10.5-bin.tar.gz \
+  && tar xvfz apache-ant-1.10.5-bin.tar.gz -C /opt \
+  && rm apache-ant-1.10.5-bin.tar.gz \
+  && ln -s /opt/apache-ant-1.10.5 /opt/ant \
+  && ln -s /opt/ant/bin/ant /usr/bin/ant \
+  && cd /opt/ant/lib \
+  && wget https://sourceforge.net/projects/ant-contrib/files/ant-contrib/1.0b3/ant-contrib-1.0b3-bin.tar.gz \
+  && tar xfvz ant-contrib-1.0b3-bin.tar.gz -C .\
+  && rm ant-contrib-1.0b3-bin.tar.gz \
+  && cd ant-contrib \
+  && mv ant-contrib-1.0b3.jar .. \
+  && cd .. \
+  && rm -rf ant-contrib
+
+# Install Cuda version 7.5.18
+RUN cd /usr/src/ \
+  && wget http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/cuda_7.5.18_linux.run \
+  && chmod +x cuda_7.5.18_linux.run \
+  && /usr/src/cuda_7.5.18_linux.run --silent --toolkit --override
+
+# add user home/jenkins
+RUN useradd -ms /bin/bash ${USER} \
+  && mkdir /home/${USER}/.ssh/
+COPY authorized_keys /home/${USER}/.ssh/authorized_keys
+COPY known_hosts /home/${USER}/.ssh/known_hosts
+RUN chown -R ${USER}:${USER} /home/${USER} \
+  && chmod 644 /home/${USER}/.ssh/authorized_keys
```

Comment:
What default permissions will the authorized keys get if we don't change it? Do we need to change perms on known hosts as well?

Reply:
The permissions will be the same as the file that was copied from, but the permissions for both authorized_keys and known_hosts need to be 644 (rw-r--r--)

Reply:
Ok so we should add a line for known hosts too then.